### PR TITLE
Thread and Mixed Platform Support

### PR DESCRIPTION
**Pull request by mitsuhiko**
This is a work in progress pull request that implements mixed platforms on a frame by frame basis as well as support for handling multiple threads and stacktraces. Among other things it also adds the baseline support for the new flexible symbolication support which is in #3634.
## New Thread Rendering

Stacktrace only:

screen shot 2016-07-02 at 15 03 18

Exception:

screen shot 2016-07-02 at 15 03 07
## Dropdown

screen shot 2016-07-02 at 15 03 56
## Other things

The tag selector also uses lozenges now:

screen shot 2016-07-02 at 12 19 36

### CODE REVIEW

File: src/sentry/interfaces/threads.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+
+class Threads(Interface):
+    score = 1000
+
+    @classmethod
+    def to_python(cls, data):
+        threads = []
+
+        for thread in data.get('list') or ():
```
Comment (dcramer) [inline src/sentry/interfaces/threads.py]: Can we call this values instead of list

Comment (mitsuhiko) [inline src/sentry/interfaces/threads.py]: I was considering it. I just want to discourage people to send this as a list.

File: src/sentry/event_manager.py [hunk 2]
```diff
@@ -326,8 +326,6 @@ def normalize(self):
             except Exception:
                 # XXX: we should consider logging this.
                 pass
-            else:
-                data['tags'].extend(inst.iter_tags())
```
Comment (dcramer) [inline src/sentry/event_manager.py]: assuming this was moved somewhere, but im not familiar with the code so just checking it was intentional

**Comment by dcramer** [comment]
I think we should improve some visuals here, but we could do that after merging.

I *am* a bit concerned about the overall event size. Do we know the rough size of a real-world payload with threads?

File: src/sentry/event_manager.py (hunk 2 above)
Comment (mitsuhiko) [inline src/sentry/event_manager.py]: Yes, this was accidentally left in which meant that tags in some circumstances were recorded twice.

**Comment by mitsuhiko** [comment]
I refactored the thread and exception rendering so that we can implement what we have in the mockup. I did not fully implement the mockup yet because I don't have the right component available to do the thread selector as @ckj had it but I think this might be a reasonable start.

**Comment by mitsuhiko** [comment]
As far as I'm concerned this is ready for merging.

**Comment by dcramer** [comment]
I think this is good, though I'm not sure the pill solution is the best for this (as its semi unstructured and variable width). I think it'd also be nice to keep the entire thread title available (rather than just the short title) outside of the select-open view. If the screenshots are up to date it also looks like there's a bit of a margin issue. I think we can deal with these when we implement the sidebar version.

**Comment by mitsuhiko** [comment]
Closing this for #3634 which has all the things now.

**Pull request closed without merging**